Jellyfin.Api/Controllers/SplashscreenController.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Jellyfin.Api.Attributes;
+using Jellyfin.Data.Enums;
+using MediaBrowser.Common.Configuration;
+using MediaBrowser.Controller.Drawing;
+using MediaBrowser.Controller.Dto;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Persistence;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Net;
+using MediaBrowser.Model.Querying;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Api.Controllers
+{
+    /// <summary>
+    /// Splashscreen controller.
+    /// </summary>
+    [Route("Splashscreen")]
+    public class SplashscreenController : BaseJellyfinApiController
+    {
+        private readonly IImageEncoder _imageEncoder;
+        private readonly IItemRepository _itemRepository;
+        private readonly IApplicationPaths _appPaths;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SplashscreenController"/> class.
+        /// </summary>
+        /// <param name="imageEncoder">Instance of the <see cref="IImageEncoder"/> interface.</param>
+        /// <param name="itemRepository">Instance of the <see cref="IItemRepository"/> interface.</param>
+        /// <param name="applicationPaths">Instance of the <see cref="IApplicationPaths"/> interface.</param>
+        /// <param name="logger">Instance of the <see cref="ILogger"/> interface.</param>
+        public SplashscreenController(
+            IImageEncoder imageEncoder,
+            IItemRepository itemRepository,
+            IApplicationPaths applicationPaths,
+            ILogger<SplashscreenController> logger)
+        {
+            _imageEncoder = imageEncoder;
+            _itemRepository = itemRepository;
+            _appPaths = applicationPaths;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Generates or gets the splashscreen.
+        /// </summary>
+        /// <param name="darken">Darken the generated image.</param>
+        /// <param name="regenerate">Whether to regenerate the image, regardless if one already exists.</param>
+        /// <returns>The splashscreen.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
```

Comment:
I don't think we add the 500 errors normally as they should not happen. Not sure though.